### PR TITLE
Invalid Loss Stopping callback

### PR DESCRIPTION
**Pull request by VolodyaCO**
This PR implements the invalid loss stopping callback and tests it, as required in #1239

### CODE REVIEW

File: netket/callbacks/invalidlossstopping.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        Args:
+            step: An integer corresponding to the step (iteration or epoch) in training.
+            log_data: A dictionary containing log data for training.
+            driver: A NetKet variational driver.
+
+        Returns:
+            A boolean. If True, training continues, else, it does not.
+        """
+        loss = np.real(getattr(log_data[driver._loss_name], self.monitor))
+        if np.isnan(loss):
```
Comment (femtobit) [inline netket/callbacks/invalidlossstopping.py:47]:
```suggestion
        if np.isnan(loss):
```
I wonder if `np.isfinite` would be better? It tests for `NaN`, but also for `±Infinity`, which also seems like a problematic expectation that might warrant aborting the simulation.

What do you think? @VolodyaCO @PhilipVinc

Comment (VolodyaCO) [reply netket/callbacks/invalidlossstopping.py:47]: Sure thing!
```suggestion
        if np.isfinite(loss):
```

File: netket/callbacks/invalidlossstopping.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        Args:
+            step: An integer corresponding to the step (iteration or epoch) in training.
+            log_data: A dictionary containing log data for training.
+            driver: A NetKet variational driver.
+
+        Returns:
+            A boolean. If True, training continues, else, it does not.
+        """
+        loss = np.real(getattr(log_data[driver._loss_name], self.monitor))
+        if np.isfinite(loss):
```
Comment (femtobit) [inline netket/callbacks/invalidlossstopping.py:47]:
```suggestion
        if not np.isfinite(loss):
```

**Review by femtobit — APPROVED** [review]
Looks good to me now. Thanks @VolodyaCO.
